x/interchainstaking/keeper/msg_server.go

```diff
@@ -130,23 +131,31 @@ func (k msgServer) RegisterZone(goCtx context.Context, msg *types.MsgRegisterZon
 
 	k.SetZone(ctx, zone)
 
+	appVersion := string(icatypes.ModuleCdc.MustMarshalJSON(&icatypes.Metadata{
```

Comment:
Given that the only thing we change in appVersion is connectionId and counterpartyConnectionId, why not create this inside the existing registerInterchainAccount method, rather than defining it separately for each instance?

Reply:
agreed

Reply:
make sense, Im checking it now